store/tikv/gcworker/gc_worker.go

```diff
@@ -530,25 +651,79 @@ func doGC(ctx goctx.Context, store tikv.Storage, safePoint uint64, identifier st
 	ticker := time.NewTicker(gcJobLogTickInterval)
 	defer ticker.Stop()
 
-	bo := tikv.NewBackoffer(tikv.GcOneRegionMaxBackoff, ctx)
+	// create task queue and task result queue, and start task workers
+	gcTaskCh := make(chan *gcTask, concurrency)
+	gcResultCh := make(chan *gcResult, concurrency*2)
+	for i := 0; i < concurrency; i++ {
+		w := newGCTaskWorker(store, gcTaskCh, gcResultCh)
+		go w.run()
+	}
+
+	var lastErr error
 	var key []byte
+GCLoop:
 	for {
 		select {
 		case <-ctx.Done():
-			return errors.New("[gc worker] gc job canceled")
+			sendQuitToTaskWorkers(gcTaskCh, concurrency)
+			lastErr = errors.New("[gc worker] gc job canceled")
+			break GCLoop
```

Comment:
This will make it dead loop forever....
When `ctx.Done()`, it will always get the signal

Reply:
ok